### ISSUE TITLE
Space-separated classes are not replaced by 'bind'

### ISSUE THREAD

**Issue by dstarcev** [JedWatson/classnames#120]
We use CSS modules in our project and recently run into an issue that the following test case was failing, which, perhaps, is not the expected behavior:

```
it('supports space separated classes', function () {
    assert.equal(classNamesBound('a b c'), '#a #b #c');
    assert.equal(classNamesBound('a   b', ['c x', ['e z']]), '#a #b #c x #e z');
});
```

I am not sure if it is not supported for performance reasons, and performance actually does degrade if we support this case. The fastest implementation I've been able to come up with so far is about 2x slower for normal strings and 4x slower for space-separated ones:
```
* local/bind#strings x 1,868,547 ops/sec ±1.46% (90 runs sampled)
*   npm/bind#strings x 3,177,634 ops/sec ±1.24% (90 runs sampled)

* local/bind#joint strings x 1,830,837 ops/sec ±1.37% (88 runs sampled)
*   npm/bind#joint strings x 6,974,965 ops/sec ±1.37% (92 runs sampled)
``` 

But it is still pretty fast and works as most users may expect.
I would like to know your opinions about the subject.

The PR is also there https://github.com/JedWatson/classnames/pull/119

**Comment by dcousens** [comment]
I don't think this would be expected behaviour for the users of `classnames`,  nor do I personally understand the intent of the PR or feature.

Going to close,  but,  happy to continue discussion.

**Comment by dcousens** [comment]
@dstarcev having re-acquainted myself with the `bind` version now,  could you explain what the failing behaviour was,  then what the expected behaviour is?

**Comment by pushred** [comment]
I encountered this in the process of adopting CSS modules for classes that are likely to conflict. I'm using Tachyons for those less likely. I looked to classnames to simplify interpolating the module classes.

Based on the [bind example](https://github.com/JedWatson/classnames#alternate-bind-version-for-css-modules) I expected classnames to work like usual but swapping module values wherever they exist for matching references in the string:

```
<h2 className={cx('f7 fw6 mt0 mb2 heading-color heading-font')}>...</h2>
```

That string is rendered as-is however. To get the module values I need to do this:

```
<h2 className={cx('f7 fw6 mt0 mb2', 'heading-color', 'heading-font')}>...</h2>
```

That works fine and the extra syntax noise isn't the worst. But it is odd in the sense that these other classes are "special". It sure beats `styles['heading-color']` but intuitively I was expecting a little more magic.
